hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnection.java

```diff
@@ -246,6 +249,15 @@ public void close(String reason, Throwable cause) {
 
         lifecycleListener.onConnectionClose(this, null, false);
         serverContext.onDisconnect(remoteAddress, cause);
+
+        LoginContext lc = (LoginContext) attributeMap.remove(LoginContext.class);
+        if (lc != null) {
+            try {
+                lc.logout();
```

Comment:
Do we know how much time a logout can take?

If we don't know, then probably it is best to get this offloaded to some executor to prevent blocking this thread for an underermined amount of time.

Reply:
Logout usually means just removing some principal instances from a `Set` in the JAAS Subject.
The member login is not offloaded either, so let's keep it here the simple way.

Reply:
If it is just removal of a set, I'm fine with it. But if it potentially involved alien code and perhaps even doing remote calls.....